pkg/model/host.go

```diff
@@ -0,0 +1,91 @@
+// Copyright 2019 The OpenSDS Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+This module implements the host data structure.
+
+*/
+
+package model
+
+import "encoding/json"
+
+// HostSpec is a comupter system which can be discoveried manually in Agentless
+// mode or automatically in Agent mode.
+// It's a consumer of volume or file share from storage.
+type HostSpec struct {
```

Comment:
For the Required fields, Can we add required: "true"?

Reply:
there is no better way to control required field. But we can add api validation to solve it.

Reply:
okay